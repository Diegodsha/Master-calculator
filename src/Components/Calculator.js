import React, { useState } from 'react';
import styled from 'styled-components';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../Logic/Calculate';

const CalcWrapper = styled.div`
  background-color: black;
  padding: 10px;
  border-radius: 5px;
  max-width: 700px;
`;

export const TextWithShadow = styled.p`
text-shadow: 2px 2px 2px black;
`;

const Calculator = () => {
  const [State, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setState(calculate(State, buttonName));
  };

  return (
    <div className="row g-0 justify-content-center justify-content-lg-between">
      <div className="col-12 col-md-5 d-flex justify-content-center align-items-center">
        <TextWithShadow className="display-1 text-center text-white  ">Let&apos;s do some math</TextWithShadow>
      </div>
      <CalcWrapper className="col-12 col-md-7 ">
        <Display
          result={State.next || State.total || '0'}
          operation={State.operation}
        />
        <ButtonPanel clickHandler={handleClick} />
      </CalcWrapper>
    </div>
  );
};

export default Calculator;
