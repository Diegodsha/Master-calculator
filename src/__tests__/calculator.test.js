import renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

describe('Calculator component', () => {
  it('Renders correctly calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
