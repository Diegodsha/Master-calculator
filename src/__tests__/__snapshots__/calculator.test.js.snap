// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Renders correctly calculator component 1`] = `
<div
  className="row g-0 justify-content-center justify-content-lg-between"
>
  <div
    className="col-12 col-md-5 d-flex justify-content-center align-items-center"
  >
    <p
      className="sc-hKFxyN LYQFs display-1 text-center text-white  "
    >
      Let's do some math
    </p>
  </div>
  <div
    className="sc-dlnjwi kBvmwR col-12 col-md-7 "
  >
    <div
      className="sc-bdnxRM dynBDI"
    >
      0
    </div>
    <div
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <button
        className="sc-gtsrHT hoqpQx"
        color="#f72020"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="X"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="sc-gtsrHT jDqPMk"
        color="#d7d7d7"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      style={
        Object {
          "height": "100px",
        }
      }
    >
      <button
        className="sc-gtsrHT bbIKaW"
        color="#d7d7d7"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="sc-gtsrHT HZmV"
        color="orange"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
